include/seqan3/search/search.hpp

```diff
@@ -138,43 +138,35 @@ struct search_configuration_validator
     /*!\brief Validates the error configuration.
      *
      * \tparam configuration_t The type of the search configuration.
+     * \tparam queries_t       The type of the search queries; must model std::ranges::forward_range.
      *
-     * \param[in] cfg The configuration to validate.
+     * \param[in] cfg     The configuration to validate.
+     * \param[in] queries The search queries.
      *
      * \throws std::invalid_argument
      *
      * \details
      *
      * Checks if the given thresholds for the configured errors are valid. Otherwise throws std::invalid_argument.
      */
-    template <typename configuration_t>
-    static void validate_error_configuration(configuration_t const & cfg)
+    template <typename configuration_t, std::ranges::forward_range queries_t>
+    static void validate_error_configuration(configuration_t const & cfg, queries_t && queries)
     {
         static_assert(detail::is_type_specialisation_of_v<configuration_t, configuration>,
                       "cfg must be a specialisation of seqan3::configuration.");
 
-        using search_traits_t = detail::search_traits<configuration_t>;
-
-        if constexpr (search_traits_t::search_with_max_error)
+        for (auto query : queries)
```

Comment:
No, we don't want that here. This should be then checked within the algorithm. If you have to process 10 million reads you have to go over them already twice. This is really nothing you want to do. The validation check should be constant. Also you don't want to expose the policy behavior, since it is a specific behavior of the algorithm, which can change for other algorithms. So this would be not generic anymore.